utils/files.ts

```diff
@@ -98,3 +118,165 @@ export function streamToCSV (
     throw new Error(RESPONSE_MESSAGES.COULD_NOT_DOWNLOAD_FILE_500.message)
   }
 }
+
+export const collapseSmallPayments = (
+  payments: TransactionsWithPaybuttonsAndPrices[],
+  currency: SupportedQuotesType,
+  timezone: string,
+  collapseThreshold: number
+): TransactionFileData[] => {
+  const treatedPayments: TransactionFileData[] = []
+  const tempGroups: Record<string, TransactionsWithPaybuttonsAndPrices[]> = {}
+  let totalPaymentsTreated = 0
+
+  const pushTempGroup = (groupKey: string): void => {
+    const tempGroup = tempGroups[groupKey]
+    if (tempGroup === undefined || tempGroup.length === 0) return
+    if (tempGroup.length === 1) {
+      pushTx(tempGroup[0])
+      tempGroups[groupKey] = []
+      return
+    }
+    const totalAmount = tempGroup.reduce((sum, p) => sum + Number(p.amount), 0)
+    const totalValue = tempGroup.reduce((sum, p) => sum + Number(getTransactionValue(p)[currency]), 0)
+    const rate = totalValue / totalAmount
```

Comment:
Same thing  I commented below about float precision error, with the difference that here we have many txs. These txs, however, all belong to the same UTC day, so they should have the same price. What you could do is

```
const uniquePrices = new Set(
  tempTxGroup
    .map(tx => tx.prices.find(p => p.price.quoteId === QUOTE_IDS[currency.toUpperCase()])!.price.value)
);
```

and then 
```
if (uniquePrices.length !== 1) {
// Should never come here
throw new Error(RESPONSE_MESSAGES.500_INVALID_PRICES_FOR_TX_ON_CSV_CREATION(uniquePrices.length))
}

const rate = uniquePrices[0]
```